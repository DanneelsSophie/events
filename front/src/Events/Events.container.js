import PropTypes from 'prop-types';

import React from 'react';
import Events from './Events';
import { useModal, useGetEvents } from './Events.hook';

function EventsContainer({ useModalFn, useGetEventsFn, EventsCmpt }) {
  const { open, handleOpen, handleClose } = useModalFn({});
  const { refetch, data, messageApi } = useGetEventsFn({});
  console.log(data);
  return (
    <EventsCmpt
      messageApi={messageApi}
      refetchGetEvents={refetch}
      openFormModalCt={open}
      openFormModalFn={handleOpen}
      closeFormModalFn={handleClose}
      events={data ?? []}
    />
  );
}

EventsContainer.propTypes = {
  useModalFn: PropTypes.func,
  useGetEventsFn: PropTypes.func,
  EventsCmpt: PropTypes.func,
};

EventsContainer.defaultProps = {
  useModalFn: useModal,
  useGetEventsFn: useGetEvents,
  EventsCmpt: Events,
};

export default EventsContainer;
