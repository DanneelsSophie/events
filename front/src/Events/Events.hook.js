import { useState } from 'react';
import { useQuery } from 'react-query';
import fetch from 'cross-fetch';

import {
  SERVICE_NAME,
  API_ERROR_MESSAGE,
  MESSAGE_LOADING,
  MESSAGE_ERROR,
  STATUS_KEY,
} from './constants';

export const getEvents = async ({ fetchFn = fetch, urlApi = process.env.REACT_APP_BASE_API_URL }) =>
  fetchFn(`${urlApi}/events`, {
    method: 'GET',
  }).then(res => {
    if (!res.ok) {
      throw new Error(API_ERROR_MESSAGE);
    }
    return res.json();
  });

export const getMessageApi = (isError, isLoading) => {
  if (isLoading) return MESSAGE_LOADING;
  if (isError) return MESSAGE_ERROR;
  return '';
};

export const getStatus = ({ startDate, endDate }) => {
  if (new Date(endDate) < new Date()) {
    return STATUS_KEY.PAST;
  }
  if (new Date(startDate) > new Date()) {
    return STATUS_KEY.NEXT;
  }
  return STATUS_KEY.CURRENT;
};

export const useGetEvents = ({ useQueryFn = useQuery, getMessageApiFn = getMessageApi }) => {
  const { data, refetch, isError, isLoading } = useQueryFn(SERVICE_NAME, getEvents);
  const messageApi = getMessageApiFn(isError, isLoading);

  return {
    data: data?.map(event => ({ ...event, status: getStatus(event) })),
    refetch,
    messageApi,
  };
};

export const useModal = () => {
  const [open, setOpen] = useState(false);
  const handleOpen = () => setOpen(true);
  const handleClose = () => setOpen(false);
  return { handleClose, handleOpen, open };
};
