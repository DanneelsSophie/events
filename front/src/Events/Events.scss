.events {
  &__list__event {
    height: 100%;
    width: 95%;
    border: solid;
    margin: 10px;
  }
  &__button {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    margin-right: 10%;
    margin-bottom: 5%;
    margin-top: 5%;
  }
}
