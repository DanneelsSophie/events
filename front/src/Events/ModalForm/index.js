export { default } from './ModalForm.container';
