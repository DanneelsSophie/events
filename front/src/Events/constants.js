export const SERVICE_NAME = 'getEvents';
export const TITLE = 'Kumojin';

export const STATUS = {
  CURRENT: { text: 'en cours', color: 'primary' },
  NEXT: { text: 'futur', color: 'success' },
  PAST: { text: 'pass√©', color: 'secondary' },
};

export const STATUS_KEY = {
  CURRENT: 'CURRENT',
  PAST: 'PAST',
  NEXT: 'NEXT',
};

export const API_ERROR_MESSAGE = 'Veuillez contacter le service technique !';
export const MESSAGE_ERROR =
  "Si vous voyez ce message, veuillez contacter le service technique. L'application n'est pas fonctionnelle";
export const MESSAGE_LOADING = 'Loading ...';
