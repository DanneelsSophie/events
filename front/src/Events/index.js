export { default } from './Events.container';
