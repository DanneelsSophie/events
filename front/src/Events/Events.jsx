import React from 'react';
import PropTypes from 'prop-types';
import AddIcon from '@mui/icons-material/Add';
import {
  AppBar,
  CardContent,
  Typography,
  Card,
  Toolbar,
  Button,
  CardMedia,
  Chip,
  Stack,
} from '@mui/material';
import { fr } from 'date-fns/locale';
import { format } from 'date-fns';

import ModalForm from './ModalForm';
import { TITLE, STATUS } from './constants';
import './Events.scss';

const CardEvent = ({ name, description, startDate, endDate, status }) => (
  <Card className="events__list__event">
    <CardContent sx={{ display: 'flex' }}>
      <CardMedia
        component="img"
        sx={{ minWidth: '100px', width: '20%', marginRight: '10px' }}
        image="./event.jpg"
        alt="image évènement"
      />
      <Stack direction="column" spacing={2}>
        <Typography gutterBottom variant="h4" component="div">
          {name}
          <Chip
            className="events__list__event__badge"
            color={STATUS[status].color}
            label={STATUS[status].text}
            size="small"
          />
        </Typography>

        <Typography gutterBottom variant="h6" component="div">
          du {format(new Date(startDate), 'eee dd MMM yyyy à HH:mm (OOOO) ', { locale: fr })}
          au {format(new Date(endDate), 'eee dd MMM yyyy à HH:mm (OOOO)', { locale: fr })}
        </Typography>
        <Typography variant="body2" color="text.secondary" align="center">
          {description}
        </Typography>
      </Stack>
    </CardContent>
  </Card>
);

CardEvent.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  startDate: PropTypes.string.isRequired,
  endDate: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
};

const Events = ({
  messageApi,
  events,
  openFormModalCt,
  openFormModalFn,
  closeFormModalFn,
  refetchGetEvents,
  ModalFormCmpt,
}) => (
  <div className="events">
    <AppBar position="static">
      <Toolbar>
        <Typography variant="h6">{TITLE}</Typography>
      </Toolbar>
    </AppBar>
    {messageApi !== '' ? (
      <h1>{messageApi}</h1>
    ) : (
      <div>
        <div className="events__list">
          {events &&
            events.map(({ _id, name, description, startDate, endDate, status }) => (
              <CardEvent
                key={_id}
                status={status}
                name={name}
                description={description}
                startDate={startDate}
                endDate={endDate}
              />
            ))}
        </div>
        <ModalFormCmpt
          openModalForm={openFormModalCt}
          handleCloseModalFormFn={closeFormModalFn}
          refetchGetEvents={refetchGetEvents}
        />
        <div className="events__button">
          <Button onClick={openFormModalFn} endIcon={<AddIcon />} variant="contained">
            Ajouter un évènement
          </Button>
        </div>
      </div>
    )}
  </div>
);

Events.propTypes = {
  messageApi: PropTypes.string.isRequired,
  events: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string,
      name: PropTypes.string,
      description: PropTypes.string,
      startDate: PropTypes.string,
      endDate: PropTypes.string,
      status: PropTypes.string,
    }),
  ).isRequired,
  openFormModalCt: PropTypes.bool.isRequired,
  openFormModalFn: PropTypes.func.isRequired,
  closeFormModalFn: PropTypes.func.isRequired,
  refetchGetEvents: PropTypes.func.isRequired,
  ModalFormCmpt: PropTypes.func,
};
Events.defaultProps = { ModalFormCmpt: ModalForm };

export default Events;
