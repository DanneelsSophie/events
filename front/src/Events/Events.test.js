import { render, screen, getByText, fireEvent } from '@testing-library/react';
import timezoneMock from 'timezone-mock';

import Events from './Events';

describe('<Events/>', () => {
  const openFormModalCt = true;
  const openFormModalFn = jest.fn();
  const closeFormModalFn = jest.fn();
  const refetchGetEvents = jest.fn();
  function ModalFormCmpt() {
    return <p>ModalForm</p>;
  }

  beforeAll(() => {
    timezoneMock.register('US/Pacific');
  });

  afterAll(() => {
    timezoneMock.unregister();
  });

  it('should Events render when API is KO or Loading displays a message', () => {
    const { baseElement } = render(
      <Events
        events={[]}
        messageApi="ERROR"
        openFormModalCt={openFormModalCt}
        openFormModalFn={openFormModalFn}
        closeFormModalFn={closeFormModalFn}
        refetchGetEvents={refetchGetEvents}
        ModalFormCmpt={ModalFormCmpt}
      />,
    );
    expect(baseElement).toMatchSnapshot();
    expect(screen.getByText('ERROR')).toBeInTheDocument();
    expect(screen.queryByText('Ajouter un évènement')).toBeNull();
  });

  it('should Events render when API is OK with empty list events', () => {
    const { baseElement } = render(
      <Events
        messageApi=""
        events={[]}
        openFormModalCt={openFormModalCt}
        openFormModalFn={openFormModalFn}
        closeFormModalFn={closeFormModalFn}
        refetchGetEvents={refetchGetEvents}
        ModalFormCmpt={ModalFormCmpt}
      />,
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should Events render when API is OK with one event', () => {
    const { baseElement } = render(
      <Events
        events={[
          {
            _id: '12',
            status: 'CURRENT',
            name: "nom de l'évènement",
            startDate: '1992-04-14T16:00:00.000Z',
            endDate: '2000-04-14T16:00:00.000Z',
            description:
              'Le lorem ipsum (également appelé faux-texte, lipsum, ou bolo bolo1) est, en imprimerie, une suite de mots sans signification utilisée à titre provisoire pour calibrer une mise en page, le texte défiee',
          },
        ]}
        messageApi=""
        openFormModalCt={openFormModalCt}
        openFormModalFn={openFormModalFn}
        closeFormModalFn={closeFormModalFn}
        refetchGetEvents={refetchGetEvents}
        ModalFormCmpt={ModalFormCmpt}
      />,
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should Events render when API is OK with three events', () => {
    const { baseElement } = render(
      <Events
        events={[
          {
            _id: '0',
            status: 'PAST',
            name: "nom de l'évènement",
            startDate: '2000-04-14T16:00:00.000Z',
            endDate: '2020-04-14T16:00:00.000Z',
            description:
              'Le lorem ipsum (également appelé faux-texte, lipsum, ou bolo bolo1) est, en imprimerie, une suite de mots sans signification utilisée à titre provisoire pour calibrer une mise en page, le texte défiee',
          },
          {
            _id: '1',
            status: 'NEXT',
            name: "nom de l'évènement 1",
            startDate: '2001-04-14T16:00:00.000Z',
            endDate: '2021-04-14T16:00:00.000Z',
            description:
              'Le lorem ipsum (également appelé faux-texte, lipsum, ou bolo bolo1) est, en imprimerie, une suite de mots sans signification utilisée à titre provisoire pour calibrer une mise en page, le texte défiee',
          },
          {
            _id: '2',
            status: 'CURRENT',
            name: "nom de l'évènement 1",
            startDate: '2001-04-14T16:00:00.000Z',
            endDate: '2030-04-14T16:00:00.000Z',
            description:
              'Le lorem ipsum (également appelé faux-texte, lipsum, ou bolo bolo1) est, en imprimerie, une suite de mots sans signification utilisée à titre provisoire pour calibrer une mise en page, le texte défiee',
          },
        ]}
        messageApi=""
        openFormModalCt={openFormModalCt}
        openFormModalFn={openFormModalFn}
        closeFormModalFn={closeFormModalFn}
        refetchGetEvents={refetchGetEvents}
        ModalFormCmpt={ModalFormCmpt}
      />,
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should call function openFormModalFn when the user clicks on the button  Ajouter un évènement', () => {
    render(
      <Events
        messageApi=""
        events={[]}
        openFormModalCt={openFormModalCt}
        openFormModalFn={openFormModalFn}
        closeFormModalFn={closeFormModalFn}
        refetchGetEvents={refetchGetEvents}
        ModalFormCmpt={ModalFormCmpt}
      />,
    );

    fireEvent.click(screen.getByText(/Ajouter un évènement/i));
    expect(openFormModalFn).toHaveBeenCalled();
  });
});
