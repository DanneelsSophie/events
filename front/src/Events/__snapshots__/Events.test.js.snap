// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Events/> should Events render when API is KO or Loading displays a message 1`] = `
<body>
  <div>
    <div
      class="events"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Kumojin
          </h6>
        </div>
      </header>
      <h1>
        ERROR
      </h1>
    </div>
  </div>
</body>
`;

exports[`<Events/> should Events render when API is OK with empty list events 1`] = `
<body>
  <div>
    <div
      class="events"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Kumojin
          </h6>
        </div>
      </header>
      <div>
        <div
          class="events__list"
        />
        <p>
          ModalForm
        </p>
        <div
          class="events__button"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Ajouter un évènement
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`<Events/> should Events render when API is OK with one event 1`] = `
<body>
  <div>
    <div
      class="events"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Kumojin
          </h6>
        </div>
      </header>
      <div>
        <div
          class="events__list"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root events__list__event css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-1u0k5eo-MuiCardContent-root"
            >
              <img
                alt="image évènement"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-efkekc-MuiCardMedia-root"
                src="./event.jpg"
              />
              <span
                class="MuiBadge-root events__list__event__badge css-1c32n2y-MuiBadge-root"
              >
                <span
                  class="MuiBadge-anchorOriginBottomRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginBottomRight css-1frogym-MuiBadge-badge"
                >
                  en cours
                </span>
              </span>
              <div
                class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
              >
                nom de l'évènement
                <div
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  du 
                  mar. 14 avr. 1992 à 9:00 (GMT-07:00) 
                  au 
                  ven. 14 avr. 2000 à 9:00 (GMT-07:00)
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
                >
                  Le lorem ipsum (également appelé faux-texte, lipsum, ou bolo bolo1) est, en imprimerie, une suite de mots sans signification utilisée à titre provisoire pour calibrer une mise en page, le texte défiee
                </p>
              </div>
            </div>
          </div>
        </div>
        <p>
          ModalForm
        </p>
        <div
          class="events__button"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Ajouter un évènement
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`<Events/> should Events render when API is OK with three events 1`] = `
<body>
  <div>
    <div
      class="events"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Kumojin
          </h6>
        </div>
      </header>
      <div>
        <div
          class="events__list"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root events__list__event css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-1u0k5eo-MuiCardContent-root"
            >
              <img
                alt="image évènement"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-efkekc-MuiCardMedia-root"
                src="./event.jpg"
              />
              <span
                class="MuiBadge-root events__list__event__badge css-1c32n2y-MuiBadge-root"
              >
                <span
                  class="MuiBadge-anchorOriginBottomRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginBottomRight css-12slcdh-MuiBadge-badge"
                >
                  passé
                </span>
              </span>
              <div
                class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
              >
                nom de l'évènement
                <div
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  du 
                  ven. 14 avr. 2000 à 9:00 (GMT-07:00) 
                  au 
                  mar. 14 avr. 2020 à 9:00 (GMT-07:00)
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
                >
                  Le lorem ipsum (également appelé faux-texte, lipsum, ou bolo bolo1) est, en imprimerie, une suite de mots sans signification utilisée à titre provisoire pour calibrer une mise en page, le texte défiee
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root events__list__event css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-1u0k5eo-MuiCardContent-root"
            >
              <img
                alt="image évènement"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-efkekc-MuiCardMedia-root"
                src="./event.jpg"
              />
              <span
                class="MuiBadge-root events__list__event__badge css-1c32n2y-MuiBadge-root"
              >
                <span
                  class="MuiBadge-anchorOriginBottomRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSuccess MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginBottomRight css-vnvyba-MuiBadge-badge"
                >
                  futur
                </span>
              </span>
              <div
                class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
              >
                nom de l'évènement 1
                <div
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  du 
                  sam. 14 avr. 2001 à 9:00 (GMT-07:00) 
                  au 
                  mer. 14 avr. 2021 à 9:00 (GMT-07:00)
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
                >
                  Le lorem ipsum (également appelé faux-texte, lipsum, ou bolo bolo1) est, en imprimerie, une suite de mots sans signification utilisée à titre provisoire pour calibrer une mise en page, le texte défiee
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root events__list__event css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-1u0k5eo-MuiCardContent-root"
            >
              <img
                alt="image évènement"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-efkekc-MuiCardMedia-root"
                src="./event.jpg"
              />
              <span
                class="MuiBadge-root events__list__event__badge css-1c32n2y-MuiBadge-root"
              >
                <span
                  class="MuiBadge-anchorOriginBottomRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginBottomRight css-1frogym-MuiBadge-badge"
                >
                  en cours
                </span>
              </span>
              <div
                class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
              >
                nom de l'évènement 1
                <div
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  du 
                  sam. 14 avr. 2001 à 9:00 (GMT-07:00) 
                  au 
                  dim. 14 avr. 2030 à 9:00 (GMT-07:00)
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
                >
                  Le lorem ipsum (également appelé faux-texte, lipsum, ou bolo bolo1) est, en imprimerie, une suite de mots sans signification utilisée à titre provisoire pour calibrer une mise en page, le texte défiee
                </p>
              </div>
            </div>
          </div>
        </div>
        <p>
          ModalForm
        </p>
        <div
          class="events__button"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Ajouter un évènement
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
